lib/src/native/realm_core.dart

```diff
@@ -61,25 +61,25 @@ class _RealmCore {
 
   String get libraryVersion => _realmLib.realm_get_library_version().cast<Utf8>().toDartString();
 
-  LastError? getLastError([Allocator? allocator]) {
-    if (allocator != null) {
-      final error = _realmLib.realm_get_last_error();
-
-      if (error == nullptr) {
-        return null;
-      }
-
-      String? message;
-      if (error.ref.message != nullptr) {
-        message = error.ref.message.cast<Utf8>().toDartString();
-      }
-      _realmLib.realm_release_last_error(error);
+  LastError? getLastError(Arena arena) {
+    final error = arena<realm_error_t>();
+    final success = _realmLib.realm_get_last_error(error);
+    if (!success) {
+      return null;
+    }
 
-      return LastError(error.ref.error, message);
+    String? message;
+    if (error.ref.message != nullptr) {
+      message = error.ref.message.cast<Utf8>().toDartString();
     }
 
-    return using((Arena arena) {
-      return getLastError(arena);
+    return LastError(error.ref.error, message);
+  }
+
+  void raiseLastErrorAsRealmException([String? errorMessage]) {
+    using((Arena arena) {
+      final lastError = getLastError(arena);
+      throw RealmException("${errorMessage ?? ""} ${lastError.toString()}");
```

Comment:
`lastError` can be null here. same as the old code actually

Reply:
also this will result in a bit weird error message. 
Previously it was 
```Dart
"Class $className not found in ${realm.config.path}. Error: table not found"
```
now it will be 
```Dart
"Class $className not found in ${realm.config.path} table not found`
```

Reply:
Actually LastError.toString cause it to format like: `Class $className not found in ${realm.config.path} Error Code: ${lastError.code} Message: table not found`. I will add appropriage punctuation.

Reply:
Fixed

Reply:
ok I see. resolving